### PR TITLE
fix (addon): #1113 re-enable button after customization

### LINKED ISSUE #1113: Bug 1295943 - The Activity Stream "Share" button is disabled after exiting Firefox menu costomization
via https://bugzilla.mozilla.org/show_bug.cgi?id=1295943
<br>

[Affected versions]:
- Firefox 45.0 and up
- Activity Stream 1.1.1-dev

[Affected Platforms]:
- All Windows
- All Mac OS X
- All Linux

[Prerequisites]:
- Latest Activity Stream add-on version (1.1.0-dev) was previously installed on a clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites. 
2. Visit any page.
3. Click the "Menu" button and go to "Customization".
4. Exit customization.
5. Observe and click the Activity Stream "Share" button.

[Expected result]:
- The "Share" button is clickable and if clicked the dropdown menu appears.

[Actual result]:
- The "Share" button is greyed out and is not clickable.

[Notes]:
- The "Share" button will be disabled for all opened tabs, but if you navigate to a new page it will become active.
- Attached a screen recording of the issue https://goo.gl/q9OT61.

### PR DESCRIPTION
**Pull request by rlr**
Fixes #1113 

r? @ncloudioj

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/1119)

### CODE REVIEW

File: lib/ShareProvider.js [hunk 1]
```diff
@@ -344,7 +350,7 @@ ShareProvider.prototype = {
       }
     };
 
-    CustomizableUI.createWidget(shareButton);
+    widget = CustomizableUI.createWidget(shareButton);
```
Comment (ncloudioj) [inline lib/ShareProvider.js]: Is this `widget` already declared/defined at somewhere before? I couldn't find it.

Comment (rlr) [inline lib/ShareProvider.js]: yes, right here -> https://github.com/rlr/activity-streams/blob/2ff6007bcf8c77e44b9a73da56ee5f4586b8a50f/lib/ShareProvider.js#L353

Comment (ncloudioj) [inline lib/ShareProvider.js]: missing a `let` here?

Comment (rlr) [inline lib/ShareProvider.js]: ah... it's actually declared at the top of the `_createButton()` function -> https://github.com/rlr/activity-streams/blob/2ff6007bcf8c77e44b9a73da56ee5f4586b8a50f/lib/ShareProvider.js#L231

Then re-set at the button after creating the widget.

Comment (ncloudioj) [inline lib/ShareProvider.js]: cool 👍

**Comment by ncloudioj** [comment]
r+

**Pull request merged**